sunpy/net/helioviewer.py

```diff
@@ -23,30 +23,61 @@
 
 class HelioviewerClient(object):
     """Helioviewer.org Client"""
-    def __init__(self, url="https://legacy.helioviewer.org/api/"):
+    def __init__(self, url="https://api.helioviewer.org/"):
         """
         url : location of the Helioviewer API.  The default location points to
-            version 1 of the API.  Version 1 of the Helioviewer API is
-            currently planned to be supported until the end of April 2017.
+            version 2 of the API.
         """
         self._api = url
-
+        self._source_dict = dict()
+        self.init_src_dict()
+    
+    def init_src_dict(self):
+        datasources = self.get_data_sources() # doctest: +REMOTE_DATA
+
+        for name, observ in datasources.items():  # doctest: +REMOTE_DATA
+            if name == "TRACE":  # doctest: +REMOTE_DATA
+                for instr, params in observ.items():  # doctest: +REMOTE_DATA
+                    self._source_dict[(name, instr, 'None', 'None')] = params['sourceId']  # doctest: +REMOTE_DATA
+            else:  # doctest: +REMOTE_DATA
+                for inst, detect in observ.items():  # doctest: +REMOTE_DATA
+                    for wavelength, params in detect.items():  # doctest: +REMOTE_DATA
+                        if name in ["Hinode", "STEREO_A", "STEREO_B"] or wavelength in ["C2", "C3"]:  # doctest: +REMOTE_DATA
+                            for wave, adict in params.items():  # doctest: +REMOTE_DATA
+                                self._source_dict[(name, inst, wavelength, wave)] = adict['sourceId']  # doctest: +REMOTE_DATA
+                        else:  # doctest: +REMOTE_DATA
+                            self._source_dict[(name, inst, 'None', wavelength)] = params['sourceId']  # doctest: +REMOTE_DATA
```

Comment:
@sunpy/sunpy-developers Anyone have a better way to do this? This is probably the ugliest piece of code I've written.